src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -204,15 +230,15 @@ private static ConsoleColor GetConsoleColor(MessageType messageType) =>
             _ => Console.ForegroundColor,
         };
 
-    private sealed class Message
+    private sealed class Message(MessageType messageType, string finalMessage)
     {
-        public MessageType MessageType { get; }
-        public string FinalMessage { get; }
+        public MessageType MessageType { get; } = messageType;
+        public string FinalMessage { get; } = finalMessage;
+    }
 
-        public Message(MessageType messageType, string finalMessage)
-        {
-            MessageType = messageType;
-            FinalMessage = finalMessage;
-        }
+    private readonly struct TelemetryMessage(string key, string value)
+    {
+        public string Key { get; } = key;
+        public string Value { get; } = value;
     }
```

Comment:
This is very close to a record definition without IsExternalInit. Consider using this instead, which gives resolves any equality/hashcode issues.
```cs
readonly record struct TelemetryMessage
{
    public TelemetryMessage(string key, string value)
    { 
        Key = key;
        Value = value;
    }
    
    public string Key { get; }
    public string Value { get; }
}
```